autopr/utils/repo.py

```diff
@@ -139,3 +146,29 @@ def repo_to_file_descriptors(repo: Repo, context_window: int, file_chunk_size: i
 
     _file_descriptor_cache[key] = file_descriptor_list
     return file_descriptor_list
+
+
+def is_path_ignored(path: str, ignore_patterns: list[str]) -> bool:
+    # Ensure we're working with a relative path
+    relative_path = os.path.relpath(path)
+    pathspec = PathSpec.from_lines(GitWildMatchPattern, ignore_patterns)
+    return pathspec.match_file(relative_path)
+
+
+def parse_gptignore(repo: Repo) -> list[str]:
+    gptignore_file = ".gptignore"
+    ignore_patterns = []
+
+    try:
+        gptignore_blob = repo.head.commit.tree / gptignore_file
+        gptignore_content = gptignore_blob.data_stream.read().decode()
+    except FileNotFoundError:
+        return ignore_patterns
+
```

Comment:
I thought this threw a `KeyError`, not `FileNotFoundError` on failure :o

But regardless, maybe let's get rid of the `try` block anyway, and let it raise all the way through on an error reading the content.
```suggestion
def parse_gptignore(repo: Repo, gptignore_filename: str = ".gptignore") -> list[str]:
    tree = repo.head.commit.tree
    if gptignore_filename not in tree:
        return []
    gptignore_blob = tree / gptignore_filename
    gptignore_content = gptignore_blob.data_stream.read().decode()
    
    ignore_patterns = []
```

Reply:
Removed `try` block